src/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,111 @@
+import React, {
+  useContext, useState,
+} from 'react';
+import { TodoList } from '../TodoList';
+import { TodosContext } from '../TodosContext';
+import { TodosFilter } from '../TodosFilter';
+
+export const TodoApp: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [allCheckd, setAllCheckd] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  const onInputChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const notCompletedTodos = [...todos].filter(todo => todo.completed === false);
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedValue = inputValue.trim();
+
+    if (trimmedValue) {
+      const newTodo = {
+        id: +new Date(),
+        title: trimmedValue,
+        completed: false,
+      };
+
+      setTodos(prevTodos => (
+        [...prevTodos,
+          newTodo,
+        ]
+      ));
+
+      setInputValue('');
+    }
+  };
+
+  const checkingAllHandler = () => {
+    setTodos(todos.map(t => (
```

Comment:
`t` is bad name, even here :)
use `todo`, etc.

Reply:
agree, please do it